contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol

```diff
@@ -1,75 +1,83 @@
 pragma solidity ^0.8.19;
 
-import {Functions} from "./Functions.sol";
+import {FunctionsRequest} from "./libraries/FunctionsRequest.sol";
 import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
 import {IFunctionsClient} from "./interfaces/IFunctionsClient.sol";
 
 /**
  * @title The Chainlink Functions client contract
- * @notice Contract writers can inherit this contract in order to create Chainlink Functions requests
+ * @notice Contract developers can inherit this contract in order to make Chainlink Functions requests
  */
 abstract contract FunctionsClient is IFunctionsClient {
-  IFunctionsRouter internal immutable s_router;
+  using FunctionsRequest for FunctionsRequest.Request;
+
+  IFunctionsRouter internal immutable i_router;
 
   event RequestSent(bytes32 indexed id);
   event RequestFulfilled(bytes32 indexed id);
 
-  error OnlyRouterCanFufill();
+  error OnlyRouterCanFulfill();
 
   constructor(address router) {
-    s_router = IFunctionsRouter(router);
+    i_router = IFunctionsRouter(router);
   }
 
   /**
-   * @notice Sends a Chainlink Functions request to the stored oracle address
-   * @param req The initialized Functions.Request
-   * @param subscriptionId The subscription ID
-   * @param callbackGasLimit gas limit for the fulfillment callback
-   * @return requestId The generated request ID
+   * @notice Sends a Chainlink Functions request
+   * @param request The Functions.Request initialized in memory
+   * @param subscriptionId The subscription ID that will be charged to service the request
+   * @param callbackGasLimit the amount of gas that will be available for the fulfillment callback
+   * @return requestId The generated request ID for this request
    */
   function _sendRequest(
-    Functions.Request memory req,
+    FunctionsRequest.Request memory request,
     uint64 subscriptionId,
     uint32 callbackGasLimit,
     bytes32 donId
-  ) internal returns (bytes32 requestId) {
-    bytes memory requestData = Functions.encodeCBOR(req);
-    requestId = _sendRequestBytes(requestData, subscriptionId, callbackGasLimit, donId);
+  ) internal returns (bytes32) {
+    return _sendRequestBytes(FunctionsRequest.encodeCBOR(request), subscriptionId, callbackGasLimit, donId);
   }
 
   /**
-   * @notice Sends a Chainlink Functions request to the stored oracle address
-   * @param data The initialized Functions request data
-   * @param subscriptionId The subscription ID
-   * @param callbackGasLimit gas limit for the fulfillment callback
-   * @return requestId The generated request ID
+   * @notice Sends a Chainlink Functions request
+   * @param data The CBOR encoded bytes data for a Functions request
+   * @param subscriptionId The subscription ID that will be charged to service the request
+   * @param callbackGasLimit the amount of gas that will be available for the fulfillment callback
+   * @return requestId The generated request ID for this request
    */
   function _sendRequestBytes(
```

Comment:
If we make this function publicly callable, it would make it easier to test off-chain CBOR encoding. Something to consider in an upcoming PR.

Reply:
Internal testing? The test helper contract can help with that